woocommerce/woocommerce-ios
# Store onboarding > add product action handling

Part of #8908 


## Description


⚠️ This PR is likely having merge conflicts with https://github.com/woocommerce/woocommerce-ios/pull/9033 that will be merged before this PR

This PR navigated the user to the Add Product flow after they tap on the `Add your first product` task. Notes on some related changes:

- Moved `StoreOnboardingCoordinator` from where the onboarding card collapsed state is presented (`DashboardViewController`) to within the hosting controller `StoreOnboardingViewHostingController`. The coordinator handles all navigation actions from the view `StoreOnboardingView`, including `View all` (showing expanded state which instantiates a new `StoreOnboardingViewHostingController` with its own coordinator internally), and all the tasks. The current implementation shares the same coordinator and there's a bug where the actions don't work in the expanded state because the shared coordinator is using the incorrect navigation controller (dashboard)
- In `AddProductCoordinator`, `sourceView` is updated to be optional to allow `nil` value since we can't pass the tapped task view as a `UIView` from SwiftUI - lemme know if you know any workarounds. This change affects the tablet presentation of the bottom sheet, but I think we should migrate to the iOS 15+ bottom sheet (should be feasible from my HACK Week attempt for the inventory scanner)
- In `StoreOnboardingCoordinator`, it handles the task tap action by starting `AddProductCoordinator`

## Testing instructions


- Log in if needed
- In the My store tab, tap `Add your first product` --> a bottom sheet should be shown to create a product
- Continue to publish a new product --> all the navigation flow should work smoothly

## Screenshots


<img src="https://user-images.githubusercontent.com/1945542/224641290-d49528ef-9b8e-4d20-9a01-d61784530baa.png" width="300" />


https://user-images.githubusercontent.com/1945542/224641179-0f6e1cae-fdc5-42ab-a287-2a6c448839cb.mov



---
